### PR TITLE
MODFQMMGR-161:Populate existing column values from APIs

### PR DESCRIPTION
**Pull request by kjain110**
## Purpose
_Describe the purpose of the pull request. Include background information if necessary._

## Approach
_How does this change fulfill the purpose?_

#### TODOS and Open Questions
- [ ] Use GitHub checklists. When solved, check the box and explain the answer.

## Learning
_Describe the research stage. Add links to blog posts, patterns, libraries or addons used to solve this problem._

### CODE REVIEW

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-item-details-definition.xml [hunk 1]
```diff
@@ -0,0 +1,515 @@
...
+              },
+              "queryable": true,
+              "filterValueGetter": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(material_type_ref_data.jsonb ->> 'name'::text))",
+              "valueFunction": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(:value))",
+              "valueGetter": "material_type_ref_data.jsonb ->> 'name'",
+              "visibleByDefault": false
+            },
+            {
+              "name": "item_material_type_id",
+              "dataType": {
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-item-details-definition.xml:286]: Added API's here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-item-details-definition.xml [hunk 2]
```diff
@@ -0,0 +1,515 @@
...
+              "isIdColumn": true,
+              "visibleByDefault": false
+            },
+            {
+              "name": "item_material_type",
+              "dataType": {
+                "dataType": "stringType"
+              },
+              "idColumnName": "item_material_type_id",
+              "valueSourceApi": {
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-item-details-definition.xml:269]: Added API's here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml [hunk 3]
```diff
@@ -0,0 +1,469 @@
...
+                "columnName": "item_material_type",
+                "entityTypeId": "0cb79a4c-f7eb-4941-a104-745224ae0292"
+              },
+              "filterValueGetter": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(material_type_ref_data.jsonb ->> 'name'::text))",
+              "valueFunction": "lower(${tenant_id}_mod_inventory_storage.f_unaccent(:value))",
+              "valueGetter": "material_type_ref_data.jsonb ->> 'name'",
+              "visibleByDefault": false
+            },
+            {
+              "name": "item_material_type_id",
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml:124]: Here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml [hunk 4]
```diff
@@ -0,0 +1,469 @@
...
+              "name": "item_id",
+              "dataType": {
+                "dataType": "rangedUUIDType"
+              },
+              "queryable": true,
+              "valueGetter": "item_details.id",
+              "visibleByDefault": false
+            },
+            {
+              "name": "item_material_type",
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml:103]: here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml [hunk 5]
```diff
@@ -0,0 +1,469 @@
...
+              },
+              "queryable": true,
+              "filterValueGetter": "\"left\"(lower(${tenant_id}_mod_inventory_storage.f_unaccent((item_details.jsonb -> 'status'::text) ->> 'name'::text)), 600)",
+              "valueFunction": "\"left\"(lower(${tenant_id}_mod_inventory_storage.f_unaccent(:value)), 600)",
+              "valueGetter": "item_details.jsonb -> 'status' ->> 'name'",
+              "visibleByDefault": true
+            },
+            {
+              "name": "loan_checkin_servicepoint_id",
+              "dataType": {
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml:158]: Here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml [hunk 6]
```diff
@@ -0,0 +1,469 @@
...
+              "queryable": true,
+              "valueGetter": "cispi.id",
+              "visibleByDefault": false
+            },
+            {
+              "name": "loan_checkin_servicepoint_name",
+              "source": {
+                "columnName": "service_point_name",
+                "entityTypeId": "89cdeac4-9582-4388-800b-9ccffd8d7691"
+              },
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml:179]: Here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml [hunk 7]
```diff
@@ -0,0 +1,469 @@
...
+              "queryable": true,
+              "filterValueGetter": "\"left\"(lower(src_circulation_loan.jsonb ->> 'loanDate'::text), 600)",
+              "valueFunction": "\"left\"(lower(:value), 600)",
+              "valueGetter": "src_circulation_loan.jsonb ->> 'loanDate'",
+              "visibleByDefault": false
+            },
+            {
+              "name": "loan_checkout_servicepoint_id",
+              "dataType": {
+                "dataType": "rangedUUIDType"
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml:209]: Here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml [hunk 8]
```diff
@@ -0,0 +1,469 @@
...
+                "valueJsonPath": "$.servicepoints.*.id",
+                "labelJsonPath": "$.servicepoints.*.id"
+                },
+              "queryable": true,
+              "valueGetter": "cospi.id",
+              "visibleByDefault": false
+            },
+            {
+              "name":"loan_checkout_servicepoint_name",
+              "source": {
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml:226]: Here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml [hunk 9]
```diff
@@ -0,0 +1,469 @@
...
+              "visibleByDefault": false
+            },
+            {
+              "name": "user_patron_group",
+              "source": {
+                "columnName": "group",
+                "entityTypeId": "e611264d-377e-4d87-a93f-f1ca327d3db0"
+              },
+              "valueSourceApi": {
+                "path": "groups",
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-loan-details-definition.xml:417]: Here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-user-details-definition.xml [hunk 10]
```diff
@@ -0,0 +1,430 @@
...
+              "dataType":{
+                "dataType":"stringType"
+              },
+              "queryable": true,
+              "valueGetter": "src_users_groups.jsonb ->> 'group'",
+              "filterValueGetter": "lower(${tenant_id}_mod_users.f_unaccent(src_users_groups.jsonb ->> 'group'::text))",
+              "valueFunction": "lower(${tenant_id}_mod_users.f_unaccent(:value))",
+              "visibleByDefault": false,
+              "idColumnName": "user_patron_group_id",
+              "source": {
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-user-details-definition.xml:290]: Here

File: src/main/resources/db/changelog/changes/v1.2.0/update-drv-user-details-definition.xml [hunk 11]
```diff
@@ -0,0 +1,430 @@
...
+                "valueJsonPath": "$.usergroups.*.id",
+                "labelJsonPath": "$.usergroups.*.group"
+              }
+            },
+            {
+              "name": "user_patron_group_id",
+              "dataType":{
+                "dataType":"rangedUUIDType"
+              },
+              "source": {
```
Comment (kjain110) [inline src/main/resources/db/changelog/changes/v1.2.0/update-drv-user-details-definition.xml:305]: Here

**Review by bvsharp — APPROVED** [review]

**Review by mweaver-ebsco — APPROVED** [review]

**Pull request merged**